### DISCUSSION TITLE
The 'Connect-PnPOnline' command was found in the module 'PnP.PowerShell', but the module could not be loaded

### DISCUSSION THREAD

**Discussion by lidacbl** [pnp/powershell#2967] [Q&A] [answered]
We did not have this error by using 1.12.0 pnp module but now we get it with 2.1.1 since 2023-04-01. We always Import-Module then connect to SharePoint Online for file automation tasks.

Error for "Connect-PnPOnline": The 'Connect-PnPOnline' command was found in the module 'PnP.PowerShell', but the
module could not be loaded. For more information, run 'Import-Module PnP.PowerShell'.

OS: Windows Server 2012 R2
Code:

      Import-Module -Name PnP.PowerShell

      $SharePointUrl = "https://mycompany.sharepoint.com/sites/MySite"
      $SharePointUserName = "AdminUserName"
      $SharePointPwd = Get-Content "C:\MyProject\MyPswdFile.txt" | ConvertTo-SecureString
      $SharePointCred = New-Object -TypeName System.Management.Automation.PSCredential ($SharePointUserName, $SharePointPwd)

      Connect-PnPOnline -Url $SharePointUrl -Credentials $SharePointCred

When we try to fix the issue by manually checking the code by line, we get error when running Import-Module as well.

Error for "Import-Module": Could not load file or assembly 'System.Management.Automation, Version=7.2.0.0, Culture=neutral,
PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified

**Comment by dogandponyshow** [comment, marked as answer]
Hi there [lidacbl](https://github.com/lidacbl).  I had this issue earlier after updating PS through ISE to v2.1.1  Also same error for Import-Module.


Take a look here: https://github.com/pnp/powershell/issues/2969

Release 2.1.1 has dropped support for PS 5.1 and ISE

Options are:
install PS 7.2 or later.

You could also consider rolling back to version 1.12.0:

Uninstall-Module -Name "PnP.PowerShell" -Force
Install-Module PnP.PowerShell -RequiredVersion 1.12.0 -Force

Hope that helps.

**Reply by lidacbl** [reply]
Thank you @dogandponyshow, I resolved the issue by install PS 7.3.3.
